lib/auth/vault.js

```diff
@@ -33,113 +31,79 @@ const vault = {};
 /**
  * authenticateV2Request
  *
- * @param {string} accessKey - user's accessKey
- * @param {string} signatureFromRequest - signature sent with request
- * @param {string} stringToSign - string to sign built per AWS rules
- * @param {string} algo - either SHA256 or SHA1
-* @param {RequestContext} requestContext - an instance of a RequestContext
+ * @param {string} params - the authentication parameters as returned by
+ *                          auth.extractParams
+ * @param {number} params.version - shall equal 4
+ * @param {number} params.version - shall equal 2
+ * @param {string} params.data.accessKey - the user's accessKey
+ * @param {string} params.data.signatureFromRequest - the signature read from
+ *                                                    the request
+ * @param {string} params.data.stringToSign - the stringToSign
+ * @param {string} params.data.algo - the hashing algorithm used for the
+ *                                    signature
+ * @param {string} params.data.authType - the type of authentication (query or
+ *                                        header)
+ * @param {string} params.data.signatureVersion - the version of the signature
```

Comment:
It means the same thing.  v2 is AWS and v4 is AWS4 but AWS and AWS4 is the way it is evaluated in policy conditions.  But you would have to trace through where version is being used and replace it.